pkg/migration-tool/writer/writer.go

```diff
@@ -24,49 +21,49 @@ const (
 )
 
 type RemoteWrite struct {
-	sigBlockRead  chan struct{}
-	sigBlockWrite chan struct{} // To the reader.
-	plan          *planner.Plan
-	client        remote.WriteClient
+	sigBlockRead       chan *planner.Block
+	sigBlockWrite      chan struct{} // To the reader.
+	plan               *planner.Plan
+	client             remote.WriteClient
+	progressMetricName string // Metric name to main the last pushed maxt to remote write storage.
+	migrationJobName   string // Label value to the progress metric.
 }
 
 // New returns a new remote write. It is responsible for writing to the remote write storage.
-func New(remoteWriteUrl string, plan *planner.Plan, sigRead, sigWrite chan struct{}) (*RemoteWrite, error) {
-	parsedUrl, err := url.Parse(remoteWriteUrl)
-	if err != nil {
-		return nil, fmt.Errorf("url-parse: %w", err)
-	}
-	wc, err := remote.NewWriteClient(fmt.Sprintf("writer-%d", 1), &remote.ClientConfig{
-		URL:     &config.URL{URL: parsedUrl},
-		Timeout: model.Duration(DefaultWriteTimeout),
-	})
+func New(remoteWriteUrl, progressMetricName, migrationJobName string, plan *planner.Plan, sigRead chan *planner.Block, sigWrite chan struct{}) (*RemoteWrite, error) {
+	wc, err := utils.CreateWriteClient(fmt.Sprintf("writer-%d", 1), remoteWriteUrl, model.Duration(DefaultWriteTimeout))
 	if err != nil {
 		return nil, fmt.Errorf("creating write-client: %w", err)
 	}
 	write := &RemoteWrite{
-		plan:          plan,
-		client:        wc,
-		sigBlockRead:  sigRead,
-		sigBlockWrite: sigWrite,
+		plan:               plan,
+		client:             wc,
+		sigBlockRead:       sigRead,
+		sigBlockWrite:      sigWrite,
+		migrationJobName:   migrationJobName,
+		progressMetricName: progressMetricName,
 	}
 	return write, nil
 }
 
 // Run runs the remote-writer. It waits for the remote-reader to give access to the in-memory
 // data-block that is written after the most recent fetch. After reading the block and storing
 // the data locally, it gives back the writing access to the remote-reader for further fetches.
-func (rw *RemoteWrite) Run(wg *sync.WaitGroup, readerUp *atomic.Bool) error {
+func (rw *RemoteWrite) Run() error {
 	log.Info("msg", "writer is up")
-	defer func() {
-		wg.Done()
-	}()
 	for {
-		if _, ok := <-rw.sigBlockRead; !ok {
+		blockRef, ok := <-rw.sigBlockRead
+		if !ok {
 			break
 		}
 		var buf []byte
-		blockRef := rw.plan.CurrentBlock()
-		blockRef.SetDescription("pushing...", 1)
+		blockRef.SetDescription(fmt.Sprintf("pushing %.2f...", float64(blockRef.Bytes())/float64(utils.Megabyte)), 1)
+		ps, err := utils.GetorGenerateProgressTimeseries(rw.progressMetricName, rw.migrationJobName)
+		if err != nil {
+			return fmt.Errorf("get or create progress time-series: %w", err)
```

Comment:
can be done on init and stored in plan? so here it will only be updated and appended?